test/server-unit/mysql_functions.spec.js

```diff
@@ -0,0 +1,225 @@
+    db.exec(dropForeignKeySql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#drop a no existing foreign key by calling drop_foreign_key', (done) => {
+    db.exec(dropUnknownForeignKeySql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#check if a dropped column doesnt exist', (done) => {
+    db.exec(dropFKExistSql)
+      .then(res => {
+        expect(res.length).to.equal(1);
+        expect(res[0].exist).to.equal(0);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#check if a column exists', (done) => {
+    db.exec(checkIfColumnExistSql)
+      .then(res => {
+        expect(res.length).to.equal(1);
+        expect(res[0].exist).to.equal(1);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#drop a no existing column  by calling drop_column_if_exists', (done) => {
+    db.exec(dropUnknownColumnSql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#drop an existing column  by calling drop_column_if_exists', (done) => {
+    db.exec(dropColumnSql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#check if a column has really been removed', (done) => {
+    db.exec(checkIfColumnExistSql)
+      .then(res => {
+        expect(res.length).to.equal(1);
+        expect(res[0].exist).to.equal(0);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#check if an index exists in a table', (done) => {
+    db.exec(checkIfIndexExistSql)
+      .then(res => {
+        expect(res.length).to.equal(1);
+        expect(res[0].exist).to.equal(1);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#drop a no existing index  by calling drop_index_if_exists', (done) => {
+    db.exec(dropUnknownIndexSql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#drop an existing index  by calling drop_index_if_exists', (done) => {
+    db.exec(dropIndexSql)
+      .then(() => {
+        expect(true).to.equal(true);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
+
+  it('#check if the index has really been removed', (done) => {
+    db.exec(checkIfIndexExistSql)
+      .then(res => {
+        expect(res.length).to.equal(1);
+        expect(res[0].exist).to.equal(0);
+        done();
+      })
+      .catch((error) => {
+        done(error);
+      });
+  });
```

Comment:
Could you be an `after()` method that will delete these tables if they exist?  That way they don't stick around in the developer's database?

Reply:
@jniles The changes look good to me and tests pass.   So I think it is ready to go.  Do you want to check on this last item about the after()  method?